src/main/java/org/minions/devfund/miguelcalderon/WhackAMole.java

```diff
@@ -0,0 +1,216 @@
+            int yRandomValue = new Random().nextInt(moleGrid.length);
+            place(xRandomValue, yRandomValue);
+        }
+    }
+
+    /**
+     * Whack a specified cell.
+     */
+    public void whack(int x, int y) {
+        if (stillHavingAttempts() && !giveUp(x, y)) {
+            if (moleGrid[x][y] == MOLE) {
+                moleGrid[x][y] = WHACKED_MOLE;
+                molesLeft--;
+                attemptsLeft--;
+                score++;
+            } else
+                attemptsLeft--;
+        }
+    }
+
+    /**
+     * Verifies if  player gives up.
+     *
+     * @param x axis.
+     * @param y axis.
+     * @return True if the player insert (-1, -1) otherwise false.
+     */
+    public boolean giveUp(int x, int y) {
+        return x == -1 && y == -1;
+    }
+
+    /**
+     * Verifies if user still having attempts.
+     *
+     * @return true if the rules are applied otherwise false.
+     */
+    private boolean stillHavingAttempts() {
+        return attemptsLeft > 0;
+    }
+
+    /**
+     * Shows the grid without show moles.
+     *
+     * @return StringBuilder without moles.
+     */
+
+    public StringBuilder printGridToUser() {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        for (int i = 0; i < moleGrid.length; i++) {
+            for (int j = 0; j < moleGrid.length; j++) {
+                if (moleGrid[i][j] == MOLE) {
+                    stringBuilder.append(EMPTY);
+                } else {
+                    stringBuilder.append(moleGrid[i][j]);
+                }
+            }
+            stringBuilder.append("\n");
+        }
+
+        return stringBuilder;
+    }
+
+    /**
+     * Shows the grid including moles.
+     *
+     * @return StringBuilder with entire grid including moles.
+     */
+    public StringBuilder printGrid() {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        for (int i = 0; i < moleGrid.length; i++) {
+            for (int j = 0; j < moleGrid.length; j++) {
+                stringBuilder.append(moleGrid[i][j]);
+            }
+            stringBuilder.append("\n");
+        }
+        return stringBuilder;
+    }
+
+
+    /**
+     * Verify if the cell is empty.
+     * @param x axis
+     * @param y axis
+     * @return true if it is an empty cell, otherwise false
+     */
+    public boolean isEmptyCell(int x, int y) {
+        boolean valid = false;
+        if (x >= 0 && y >= 0 && x < moleGrid.length && y < moleGrid.length)
+            valid = moleGrid[x][y] == EMPTY;
+        return valid;
+
+    }
+
+
+    public int getMolesLeft() {
+        return molesLeft;
+    }
+
+    public int getAttemptsLeft() {
+        return attemptsLeft;
+    }
+
+    public int getScore() {
+        return score;
+    }
+
+    public char[][] getMoleGrid() {
+        return moleGrid;
+    }
+
+
+
+    //Main method, test a 10x10 grid with 50 attempts.
+      /*
+    public static void main(String[] args) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Reply:
block deleted